### ISSUE TITLE
possible to use golang openai client? 

### ISSUE THREAD

**Issue by swuecho** [tmc/langchaingo#119]
https://github.com/sashabaranov/go-openai

It is pretty feature complete.  will save a lot of repetitive code .

configure change like this will be totally unnecessary. 

https://github.com/tmc/langchaingo/pull/117

**Comment by cduggn** [comment]
+1

**Comment by tmc** [comment]
Yeah I'm open to it, I started very simple here. Do you want to open a PR to suggest porting over to it?

**Comment by itsahedge** [comment]
+1

**Comment by bdqfork** [comment]
I am interested in this, you can assign it to me. @tmc

**Comment by caxqueiroz** [comment]
Hi there, 
any plans to merge #145?

**Comment by KabudoWiseMan** [comment]
+1

**Comment by Leeaandrob** [comment]
+1

**Comment by Leeaandrob** [comment]
+1

**Comment by terwey** [comment]
I didn't want to open a new issue as this one already exists, but I saw that Microsoft released a Go client for their Azure OpenAI and it's compatible with OpenAI too.

https://pkg.go.dev/github.com/Azure/azure-sdk-for-go/sdk/ai/azopenai#section-readme

I suspect it's based on their Swagger specs they documented for their deployments. I think this would be a good package to rely on as they will make sure it works.

Reason I bumped into this is because I'm using the latest GPT-4 preview on Azure and the API responses are very different than is implemented in the project right now... and instead of doing a lot of work, this could be a good alternative.

**Comment by PylotLight** [comment]
@terwey Raises a good point around officially supported solutions. 
I've been using https://github.com/sashabaranov/go-openai for my projects and it probably has more features/support I assume at this point, but long term it's worth thinking about which client we should use. 

@tmc Are you able to advise which direction you would prefer this takes for a better client to use that doesn't require this project to maintain it's own client?

https://github.com/sashabaranov/go-openai vs https://github.com/Azure/azure-sdk-for-go

Note: It's hard to beat the official MS one being updated just 4 hours ago while the unofficial was last week, although I'm sure both work just fine. Perhaps we need a list of required features or something.
